core/src/main/java/io/hyperfoil/core/session/SessionImpl.java

```diff
@@ -149,7 +149,14 @@ public Session declareObject(Object key) {
    }
 
    public Object getObject(Object key) {
-      return ((ObjectVar) requireSet(key)).get();
+      Var var = requireSet(key);
+      if (var instanceof ObjectVar) {
+         return ((ObjectVar) var).get();
+      } else if (var instanceof IntVar) {
+         return ((IntVar) var).get();
```

Comment:
I am trying to say that the step needs to be aware of the potential conversion - this is why even if it would keep the code simpler and prevent repetition of blocks like the above we won't push that from steps to the session.

Reply:
The method is `Object getObject(Object key)`. Who is calling getObject is aware of potential boxing. Or my feeling is wrong ?

Reply:
Also, `AsyncCache::CompletableFuture<V> putAsync(K key, V value)` will do the boxing even if I am doing a `put` with `int`

Reply:
Most people should not use the programmatic API but write the tests in YAML. When they create an integer and consume it as if it were an object they should get an error that will let them rethink if what they're doing is correct.

Yes, in your case the boxing is inevitable.

Reply:
Moved to `HotRodRequestStep`